test/CheckedCRewriter/allocator.c

```diff
@@ -0,0 +1,50 @@
+// Tests for Checked C rewriter tool.
+//
+// Tests for malloc and friends. 
+//
+// RUN: checked-c-convert %s -- | FileCheck -match-full-lines %s
+// RUN: checked-c-convert %s -- | %clang_cc1 -fno-builtin -verify -fcheckedc-extension -x c -
+// expected-no-diagnostics
+//
+typedef __SIZE_TYPE__ size_t;
+extern void *malloc(size_t);
+extern void free(void *);
+
+void dosomething(void) {
+  int a = 0;
+  int *b = &a;
+  *b = 1;
+  return;
+}
+
+void foo(void) {
+  int *a = (int *) malloc(sizeof(int));
+  *a = 0;
+  free(a);
+  return;
+}
+//CHECK: void foo(void) {
+//CHECK-NEXT: _Ptr<int>  a = (int *) malloc(sizeof(int));
+//CHECK-NEXT: *a = 0;
+//CHECK-NEXT: free((void *)a);
+
+typedef struct _listelt {
+  struct _listelt *next;
+  int val;
+} listelt;
+
+typedef struct _listhead {
+  listelt *hd;
+} listhead;
+
+void add_some_stuff(listhead *hd) {
```

Comment:
Yeah, that makes sense to me. If it takes me more than half an hour to figure out tomorrow, I'll make a separate work item for it and do it later.